### ISSUE TITLE
Future is not completed with anything if it throws InterruptedException

### ISSUE THREAD

**Issue by scabug** [scala/bug#8938]
When code executed as future throws InterruptedException, future is not completed (neither success or failure). Await.result will block indefinitely if timeout is infinite.

It is quite typical for any concurrent code to throw InterruptedException, for example Thread.sleep() might throw one. It is extremely error prone that this type of exception is swallowed and only indication is ExecutionContext's failure report, by default printed to console.

**Comment by SethTisue** [comment]
@viktorklang should this ticket stay open?

**Comment by viktorklang** [comment]
@SethTisue This will be solved with my improvements to Future which I will propose for 2.13

**Comment by viktorklang** [comment]
(Converting InterruptedExceptions to ExecutionExceptions and resetting the interruption flag so the EC sees it.)

**Comment by SethTisue** [comment]
> This will be solved with my improvements to Future which I will propose for 2.13

@viktorklang was this in fact resolved in 2.13?

**Comment by viktorklang** [comment]
```scala
Welcome to Scala 2.13.3 (OpenJDK 64-Bit Server VM, Java 11.0.3).
Type in expressions for evaluation. Or try :help.

scala> import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.ExecutionContext.Implicits.global

scala> import scala.concurrent._
import scala.concurrent._

scala> Future(throw new InterruptedException())
val res0: scala.concurrent.Future[Nothing] = Future(<not completed>)

scala> Await.result(res0,duration.Duration.Inf)
java.lang.InterruptedException
  at $anonfun$res1$1(<console>:1)
  at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:671)
  at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:430)
  at scala.concurrent.BatchingExecutor$AbstractBatch.runN(BatchingExecutor.scala:134)
  at scala.concurrent.BatchingExecutor$AsyncBatch.apply(BatchingExecutor.scala:163)
  at scala.concurrent.BatchingExecutor$AsyncBatch.apply(BatchingExecutor.scala:146)
  at scala.concurrent.BlockContext$.usingBlockContext(BlockContext.scala:107)
  at scala.concurrent.BatchingExecutor$AsyncBatch.run(BatchingExecutor.scala:154)
  at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
  at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
  at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
  at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
  at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
  at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```
